modules/gdscript/tests/test_gdscript_formatter.h

```diff
@@ -0,0 +1,4146 @@
+		const String code =
+				GDSCRIPT(R"(
+var my_variable := [0,1,2,3,4,5,6,7,8,9]
+)");
+		const String pre_formatted =
+				GDSCRIPT(R"(
+var my_variable := [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+)");
+
+		CHECK_FORMAT(code, pre_formatted);
+	}
+
+	TEST_CASE("Should be able to store a variable with array that wraps due to length") {
+		const String code =
+				GDSCRIPT(R"(
+var my_variable := ["Lorem ipsum dolor sit amet, consectetur adipiscing elit.","Lorem ipsum dolor sit amet, consectetur adipiscing elit.","Lorem ipsum dolor sit amet, consectetur adipiscing elit.","Lorem ipsum dolor sit amet, consectetur adipiscing elit.","Lorem ipsum dolor sit amet, consectetur adipiscing elit."]
+)");
+		const String pre_formatted =
+				GDSCRIPT(R"(
+var my_variable := [
+	"Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
+	"Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
+	"Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
+	"Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
+	"Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
+]
+)");
+
+		CHECK_FORMAT(code, pre_formatted);
+	}
+
+	TEST_CASE("Should be able to store a variable with array with subarray that wraps due to length") {
+		const String code =
+				GDSCRIPT(R"(
+var my_variable := [["Lorem ipsum dolor sit amet, consectetur adipiscing elit.","Lorem ipsum dolor sit amet, consectetur adipiscing elit.","Lorem ipsum dolor sit amet, consectetur adipiscing elit.","Lorem ipsum dolor sit amet, consectetur adipiscing elit.","Lorem ipsum dolor sit amet, consectetur adipiscing elit."]]
+)");
+		const String pre_formatted =
+				GDSCRIPT(R"(
+var my_variable := [
+	[
+		"Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
+		"Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
+		"Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
+		"Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
+		"Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
+	],
+]
+)");
+
+		CHECK_FORMAT(code, pre_formatted);
+	}
+
+	TEST_CASE("Should be able to store a variable with array that breaks, of arrays that do not break") {
+		const String code =
+				GDSCRIPT(R"(
+var my_variable := [[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10],[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10],[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]]
+)");
+		const String pre_formatted =
+				GDSCRIPT(R"(
+var my_variable := [
+	[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+	[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+	[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+]
+)");
+
+		CHECK_FORMAT(code, pre_formatted);
+	}
+
+	TEST_CASE("Should maintain type information for typed arrays") {
+		const String code =
+				GDSCRIPT(R"(
+extends Node
+
+@onready var children: Array[Node] = get_children()
+)");
+		const String pre_formatted =
+				GDSCRIPT(R"(
+extends Node
+<*noop*>
+<*noop*>
+@onready var children: Array[Node] = get_children()
+)");
+
+		CHECK_FORMAT(code, pre_formatted);
+	}
```

Comment:
I added the `GDSCRIPT` macro that improve script readability in the test file. Scripts must now begin with a `\n` when using `GDSCRIPT`.

Take note that the macro replaces the `<*noop*>` tokens with empty strings, that makes the 2+ newlines compatible with "misc/scripts/header_guards.sh"